galaxy_ng/app/dynaconf_hooks.py

```diff
@@ -709,8 +712,37 @@ def read_settings_from_cache_or_db(
 
         return temp_settings.get(key, value.value)
 
+    def alter_hostname_settings(
+        temp_settings: Settings,
+        value: HookValue,
+        key: str,
+        *args,
+        **kwargs
+    ) -> Any:
+        """Use the request headers to dynamically alter the content origin and api hostname.
+        This is useful in scenarios where the hub is accessible directly and through a
+        reverse proxy.
+        """
+
+        ALLOWED_KEYS = ['CONTENT_ORIGIN', 'ANSIBLE_API_HOSTNAME']
+
+        if not apps.ready or key.upper() not in ALLOWED_KEYS:
+            # If app is starting up or key is not on allowed list bypass and just return the value
+            return value.value
+
+        if key.upper() in ALLOWED_KEYS:
+            req = get_current_request()
+            if req is not None:
+                headers = dict(req.headers)
+                proto = headers.get('X-Forwarded-Proto', 'http')
+                host = headers.get('Host', 'localhost:5001')
+                baseurl = proto + '://' + host
+                return baseurl
+
+        return temp_settings.get(key, value.value)
```

Comment:
The `if` condition is redundant here, already checked above,
and as there is no merging of data structure involved (dicts,lists) we can
just return value, no need to build a temp_settings.

```suggestion
        
        req = get_current_request()
        if req is not None:
            headers = dict(req.headers)
            proto = headers.get('X-Forwarded-Proto', 'http')
            host = headers.get('Host', 'localhost:5001')
            baseurl = proto + '://' + host
            return baseurl
        else:
            return value.value
```

Reply:
we could just put a final `return value.value` however using the else is more readable IMO